flax/optim/base.py

```diff
@@ -215,13 +212,12 @@ class _NoAux:
   pass
 
 
-@struct.dataclass
-class Optimizer:
+class Optimizer(struct.PyTreeNode):
   """Wraps an optimizer with its hyper_params, state, and model parameters."""
 
   optimizer_def: OptimizerDef = struct.field(pytree_node=False)
-  state: Any
-  target: Any
+  state: Any = struct.field(pytree_node=True)
```

Comment:
`True` seems to be the default value so you can omit it if you want.

Reply:
Pytype things the first argument is optional due the struct.field and it then complains all the attributes that come after are required. I'll add a tracking issue for that as well